elastic/src/main/java/org/openmbee/mms/elastic/CommitElasticDAOImpl.java

```diff
@@ -127,4 +170,107 @@ protected String getIndex() {
     public CommitJson update(CommitJson commitJson) {
         return this.update(getIndex(), commitJson);
     }
+
+    private List<CommitJson> getDocs(String commitId) {
+        try {
+            QueryBuilder commitQuery = QueryBuilders.boolQuery()
+                .should(QueryBuilders.termQuery(CommitJson.ID, commitId))
+                .minimumShouldMatch(1);
+            SearchHits hits = getCommitResults(commitQuery);
+            if (hits.getTotalHits().value == 0) {
+                return new ArrayList<>();
+            }
+            List<CommitJson> rawCommits = new ArrayList<>();
+            for (SearchHit hit : hits.getHits()) {
+                CommitJson ob = new CommitJson();
+                ob.putAll(hit.getSourceAsMap());
+                rawCommits.add(ob); // gets "_source"
+            }
+            return rawCommits;
+        } catch (IOException ioe) {
+            throw new RuntimeException(ioe);
+        }
+    }
+
+    private List<CommitJson> getFullCommits(Collection<String> commitIds) {
+        return commitIds.stream().map(this::getFullCommit).collect(Collectors.toList());
+    }
+
+    private CommitJson getFullCommit(String commitId) {
+        return mungCommits(getDocs(commitId));
+    }
+
+    private SearchHits getCommitResults(QueryBuilder query) throws IOException {
+        SearchRequest searchRequest = new SearchRequest(getIndex());
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
+        sourceBuilder.query(query);
+        sourceBuilder.size(this.resultLimit); // TODO handle paging requests
+        sourceBuilder.sort(new FieldSortBuilder(CommitJson.CREATED).order(SortOrder.DESC));
+        searchRequest.source(sourceBuilder);
+        SearchResponse searchResponse = client.search(searchRequest, RequestOptions.DEFAULT);
+        return searchResponse.getHits();
+    }
+
+    private static CommitJson mungCommits(List<CommitJson> commits) {
+        return commits.stream().reduce(new CommitJson(), (partial, raw) -> {
+            if (partial.getAdded() == null) {
+                partial.setAdded(new ArrayList<>());
+            }
+            if (partial.getUpdated() == null) {
+                partial.setUpdated(new ArrayList<>());
+            }
+            if (partial.getDeleted() == null) {
+                partial.setDeleted(new ArrayList<>());
+            }
+            if (partial.getSource() == null) {
+                partial.setSource("");
+            }
+            if (partial.getComment() == null) {
+                partial.setComment("");
+            }
+            if (partial.getId() == null) {
+                partial.setId("");
+            }
+            if (partial.getId() == null) {
+                partial.setId("");
+            }
+
+            if (partial.getAdded().isEmpty() && raw.getAdded() != null) {
```

Comment:
I don't think the first check should be here? this would only add addition data if partial is empty but we want to add them regardless